include/git2/commit.h

```diff
@@ -329,6 +321,26 @@ GIT_EXTERN(int) git_commit_extract_signature(git_buf *signature, git_buf *signed
  *	the given reference will be updated to point to it
  */
 GIT_EXTERN(int) git_commit_create(
```

Comment:
So, to reiterate the API change here. The only backwards-incompatible change we're doing in this PR is to rename `git_commit_create` to `git_commit_create_on`. As this is a rather central function that is probably in use by a lot of downstream users of libgit2, this is going to break most of them. While the fix is trivial, it is going to be a huge pain e.g. for distro maintainers, who'd have to patch a whole lot of reverse-dependencies of libgit2 to keep them working.

So I'd heavily favour to have a non-breaking API change here. Keep `git_commit_create` as is and instead create a new function that handles creation of a commit without updating a reference. The pain point is that I myself have no idea what to call that function. The following list simply sucks:

- git_commit_create_v2: version numbers are rather generic and don't tell much about the function's changed intention
- git_commit_create_only: the `only` only makes sense if you know the context of updating refs
- git_commit_create_norefupdate: well, this at least says what is different from the other commit function
- git_commit_create_obj: dunno, might be just weird

I think we should either just keep `git_commit_create` untouched or mark it as deprecated. Breaking users without a proper deprecation period and without really good reasons feels irresponsible to me.